statistics/handle/internal/cache/mapcache/map_cache.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mapcache
+
+import (
+	"github.com/pingcap/tidb/statistics"
+	"github.com/pingcap/tidb/statistics/handle/internal/cache"
+)
+
+type cacheItem struct {
+	value *statistics.Table
+	key   int64
+	cost  int64
+}
+
+func (c cacheItem) copy() cacheItem {
+	return cacheItem{
+		key:   c.key,
+		value: c.value,
+		cost:  c.cost,
+	}
+}
+
+// MapCache is a cache based on map.
+type MapCache struct {
+	tables   map[int64]cacheItem
+	memUsage int64
+}
+
+// NewMapCache creates a new map cache.
+func NewMapCache() *MapCache {
```

Comment:
Better to hide the implementation, return the interface `StatsCacheInner` instead of the structure pointer `*MapCache`.

Reply:
I think it's okay here, as long as the usage is through an interface.